modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -539,6 +539,7 @@ public void processOpts() {
             apiTemplateFiles.clear();
             modelTemplateFiles.clear();
         }
+        supportsAdditionalPropertiesWithComposedSchema = true;
```

Comment:
I don't think it must be set to true by default.

Reply:
This is consistent with the way that the Java generator uses this flag when it is run with a template library that works with jackson.  I think in the case of the spring generator, all 3 libraries currently use jackson for object serialization.